watertap/unit_models/nanofiltration_DSPMDE_0D.py

```diff
@@ -0,0 +1,1394 @@
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args['feed_side'],
+            hold_state=True
+        )
+        init_log.info_high("Initialization Step 1 Complete.")
+        if not ignore_dof:
+            check_dof(blk, fail_flag=fail_on_warning, logger=init_log)
+        # ---------------------------------------------------------------------
+        # Initialize other state blocks based on properties at
+        # inlet state block
+        blk.feed_side.properties_out.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args['retentate'],)
+        blk.feed_side.properties_interface.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args['interface_in'], )
+        blk.permeate_side.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args['permeate'], )
+        blk.mixed_permeate.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args['permeate'], )
+        blk.pore_entrance.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args['interface_in'],)
+        blk.pore_exit.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args['permeate'],)
+        init_log.info_high("Initialization Step 2 Complete.")
+
+        # # Double-check for poorly scaled variables after state block initialization
+        # # and rescale them so that scaled variable values = 1:
+        badly_scaled_vars = list(iscale.badly_scaled_var_generator(blk))
+        if len(badly_scaled_vars) > 0:
+            init_log.warn(f"{len(badly_scaled_vars)} poorly scaled "
+                          f"variable(s) will be rescaled so that each scaled variable value = 1")
+            blk._automate_rescale_variables()
+        # # ---------------------------------------------------------------------
+        # # Solve unit with deactivated constraint
+        blk.eq_solute_solvent_flux.deactivate()
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+            if not check_optimal_termination(res):
+                init_log.warn("Trouble solving NanofiltrationDSPMDE0D unit model with deactivated constraint.")
+        # # ---------------------------------------------------------------------
+        # # Solve unit
+        blk.eq_solute_solvent_flux.activate()
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+            if not check_optimal_termination(res):
+                init_log.warn("Trouble solving NanofiltrationDSPMDE0D unit model. Trying one more time.")
```

Comment:
You should raise an `InitializationError` here instead. Without an exception, it is easy for this to pass unnoticed, whereas with an exception the user at least needs to acknowledge it with a `try/except` (which could just `pass` on an exception indicating they are happy to continue with the failing step).

IDAES is try to make it standard for all models to raise an `InitializationError` in these cases.